### PR TITLE
Stacki host module

### PR DESCRIPTION
**Pull request by bbyhuy**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
stacki_host

##### ANSIBLE VERSION

```
ansible 2.1.2.0
  config file = /home/vagrant/automata_modules/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

Module for managing stacki hosts (add, remove, force-install).

Referencing Revisions: [Previous PR# 19518](https://github.com/ansible/ansible/pull/19518)

```
$ ansible local -i 'localhost,' -c local -m stacki_host -a 'name=test-1 stacki_user=usr stacki_password=pwd stacki_endpoint=url prim_intf_mac=mac_addr prim_intf_ip=x.x.x.x prim_intf=eth0' 

changed: [127.0.0.1] => {"changed": true, "rc": 0, "stderr": "{}", "stdout": "api call successful", "stdout_lines": ["api call successful"]}


```

### CODE REVIEW

**Comment by bbyhuy** [comment]
@gundalow @bsanders

File: lib/ansible/modules/infrastructure/stacki/stacki_host.py
Comment (gundalow) [inline lib/ansible/modules/infrastructure/stacki/stacki_host.py]: You should mention these these env variables in the `description:`, e.g.

"If not specified the environment variable C(stacki_password) is used instead."

File: lib/ansible/modules/infrastructure/stacki/stacki_host.py
Comment (gundalow) [inline lib/ansible/modules/infrastructure/stacki/stacki_host.py]: Mention version of requests here as well

File: lib/ansible/modules/infrastructure/stacki/stacki_host.py
Comment (gundalow) [inline lib/ansible/modules/infrastructure/stacki/stacki_host.py]: Exactly 2.2.1 or 2.2.1 or higher?

**Review by gundalow — CHANGES_REQUESTED** [review]
needs_revision

Shippable is failing due to the use of requests, you'll need to add this to the skip file for that test.

**Comment by bsanders** [comment]
There's also a 'chdir' option that doesn't appear to be used anymore.

**Comment by mattclay** [comment]
@bbyhuy Tests are failing because there's no `__init__.py` in the `infrastructure` directory. That directory was recently [removed](https://github.com/ansible/ansible/commit/9816addcdc8e62ba3c2a64223fd4653bbac08250) for being too generic.

You'll need to pick a more appropriate directory for the module.

**Comment by bsanders** [comment]
This is a +1 Stacki, at least from a Stacki perspective.

@bbyhuy Do you think you'll add other stack commands, or with these two PRs are you pretty much adding what you guys will need?

**Comment by bsanders** [comment]
@bbyhuy @mattclay very recently removed, heh.  It looks like Foreman and Katello were moved to "remote_management", so Stacki modules should probably go there, too.

**Comment by bbyhuy** [comment]
@bsanders definitely will be updating it in the near future as we start adding the raiding features and network bonding.

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
If everyone is happy please respond with "ship_it" and I'll get it merged. Looks like you are, just want to check.

@bsanders Are you from Stacki? 

We define a list of module maintainers so we know who should review, I wonder if that should be you and @bbyhuy 

FYI, if in the future you start adding more modules you can pull out common code, arguments and docs into `lib/ansible/module_utils` and `docs_fragments`

**Comment by bsanders** [comment]
@gundalow Yeah, I'm one of the developers for Stacki so I suppose I should be a maintainer.  @bbyhuy is one of our power users, it probably makes sense to put him in there too, but I won't voluntell him to.

Sorry, one more thing @bbyhuy: It looks like you implement but did not document the "force_install" option.

**Comment by bbyhuy** [comment]
@gundalow I do not mind joining @bsanders as a maintainer :-)

**Comment by gundalow** [comment]
I can see the docsstrings for force_install has been updated so I'll merge this now

Maintainers added in https://github.com/ansible/ansibullbot/pull/232

Thanks!

**Pull request merged**
